src/main/java/org/truffleruby/language/exceptions/RescueNodeWrapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+
+package org.truffleruby.language.exceptions;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.instrumentation.InstrumentableFactory;
+import com.oracle.truffle.api.instrumentation.ProbeNode;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeCost;
+import com.oracle.truffle.api.object.DynamicObject;
+
+public class RescueNodeWrapper implements InstrumentableFactory<RescueNode> {
```

Comment:
It's hand-written using the same pattern as the generated code. The differences are in the call to `super` and the delegated `canHandle`.